### PR TITLE
Add JSON, YAML and XML writing and parsing support

### PR DESCRIPTION
**Pull request by xavierfigueroav**
1. Add JSON, YAML and XML writers (json.py, yaml.py and xml.py in spdx/writers/)
2. Add JSON, YAML and XML parsers and builders (jsonyamlxml.py and jsonyamlxmlbuilders.py in spdx/parsers/)

Signed-off-by: Xavier Figueroa <xavierfigueroav@gmail.com>

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -24,6 +24,7 @@ def test_suite():
         'ply',
         'rdflib',
         'six',
+        'ruamel.yaml',
```
Comment (pombredanne) [inline setup.py]: Please do not use that YAML parser: it requires a native and a C compiler. Use PyYaml instead

**Comment by xavierfigueroav** [comment]
> Just by having a quick glance, you might want to add support py3

You mean the example files only, right?

**Comment by xavierfigueroav** [comment]
@pombredanne @pombredanne I've made the changes requested. Please, let me know if something is wrong or missing

**Comment by xavierfigueroav** [comment]
@pombredanne this is ready for review. Please check!

**Comment by pombredanne** [comment]
@xavierfigueroav thank you!

**Comment by pombredanne** [comment]
I pushed a few cleanup commits. Please review

**Comment by xavierfigueroav** [comment]
@pombredanne Everything looks great!

**Comment by pombredanne** [comment]
All merged now. Thank you ++

**Pull request merged**